packages/visual-layer/src/layers/bar/bar-helper.js

```diff
@@ -65,19 +66,21 @@ const resolveDimByField = (type, axesInfo, config, data) => {
             enter = pos;
             enterSpace = space;
         } else {
-            const zeroPos = axis.getScaleValue(0);
+            const barBasePos = minDomVal < 0 ? axis.getScaleValue(0) : axis.getScaleValue(minDomVal);
             const axisType = axis.getScaleValue(data[type]);
             const axisType0 = axis.getScaleValue(data[`${type}0`]);
 
             enterSpace = 0;
             if (type === 'x') {
-                pos = data[type] < 0 || transformType === STACK ? axisType : zeroPos;
-                space = Math.abs(pos - (transformType === STACK ? axisType0 : (data[type] >= 0 ? axisType : zeroPos)));
+                pos = data[type] < 0 || transformType === STACK ? axisType : barBasePos;
```

Comment:
Lots of similar type checking. Better to use pattern.

Reply:
Done